### PR TITLE
[native] Replace hyperlink with custom markdown

### PR DESCRIPTION
**Pull request by KatPo**

### CODE REVIEW

File: native/chat/robotext-message.react.js [hunk 1]
```diff
@@ -105,7 +105,15 @@ class RobotextMessage extends React.PureComponent<Props> {
         continue;
       }
       if (splitPart.charAt(0) !== '<') {
-        textParts.push(decodeURI(splitPart));
+        textParts.push(
+          <Markdown
+            key={robotextParts.indexOf(splitPart)}
+            darkStyle={true}
```
Comment (KatPo) [inline native/chat/robotext-message.react.js:111]: Two things I'm not sure about: I've added key, since without it there was a yellow warning. I'm also passing `darkStyle={true}`, because we don't have any info about globalTheme here to pass proper value, but it's just a blue link on light/dark background and on both it looks good.
Let me know if there is a better solution about any of this!

File: native/calendar/entry.react.js [hunk 2]
```diff
@@ -392,19 +391,13 @@ class InternalEntry extends React.Component<Props, State> {
             iosActiveOpacity={opacity}
           >
             <View>
-              <View style={heightStyle} />
-              <Hyperlink
-                linkDefault={true}
-                linkStyle={linkStyle}
-                style={this.props.styles.textContainer}
+              <View style={[heightStyle, this.props.styles.textContainer]} />
```
Comment (Ashoat) [inline native/calendar/entry.react.js:394]: Why did you add `this.props.styles.textContainer` to the style here?

File: native/chat/robotext-message.react.js (hunk 1 above)
Comment (Ashoat) [reply native/chat/robotext-message.react.js:111]: Yeah, you need a key here, since you're pushing a React element onto an array. Strings are their own keys, but React elements need explicit keys specified.

1. You need to get the theme data and set `darkStyle` based on that
2. Let's make sure that the key doesn't accidentally collide with the `id` keys used below. `id` keys are numeric, so we can solve this by prefixing a string like `text${robotextParts.indexOf(splitPart)}`
3. Let's avoid calling `indexOf` here, there's no need to search through the string for an index. We can initialize a `let keyIndex = 0;` variable at the start, and just use `text${keyIndex++}` as the key

File: native/markdown/markdown.react.js [hunk 3]
```diff
@@ -6,17 +6,21 @@ import * as SimpleMarkdown from 'simple-markdown';
 import { onlyEmojiRegex } from 'lib/shared/emojis';
 
 import { getMarkdownStyles } from './styles';
-import rules from './rules.react';
+import { rules, advancedRules } from './rules.react';
 
 type Props = {|
   darkStyle: boolean,
   children: string,
+  onlyBasicRules?: boolean,
```
Comment (Ashoat) [inline native/markdown/markdown.react.js:14]: Let's pass the `rules` function in here instead of having this boolean

File: native/calendar/entry.react.js (hunk 2 above)
Comment (KatPo) [reply native/calendar/entry.react.js:394]: To make this look like before, without `this.props.styles.textContainer` it was bigger and looked like this:
Zrzut ekranu 2020-07-14 o 14 02 15
instead of 
Zrzut ekranu 2020-07-14 o 14 02 23

Comment (Ashoat) [reply native/calendar/entry.react.js:394]: That does seem like an issue, but I'm concerned that you are overriding `heightStyle` with `this.props.styles.textContainer` here. I'll look into it...

Comment (Ashoat) [reply native/calendar/entry.react.js:394]: So what you're doing here is basically making `heightStyle` useless. `textContainer` is `position: absolute`, which means this `View`'s `height` will only apply to itself.

It's important that the height of this `FlatList` item matches `heightStyle`. Otherwise our ability to calculate the position of elements in the `FlatList` could be broken.

I got it working this way:

```js
<View style={heightStyle} />
<View style={this.props.styles.textContainer}>
  <Text
    style={[this.props.styles.text, textStyle]}
    onLayout={this.onTextLayout}
  >
    <Markdown darkStyle={darkColor}>{rawText}</Markdown>
  </Text>
</View>
```

In the future, remember that it's not just about making things look correct. When you change a line of code, make sure you understand 100% why it was there before, and what it was doing.

File: native/calendar/entry.react.js [hunk 4]
```diff
@@ -392,19 +391,13 @@ class InternalEntry extends React.Component<Props, State> {
...
+                onLayout={this.onTextLayout}
               >
-                <Text
-                  style={[this.props.styles.text, textStyle]}
-                  onLayout={this.onTextLayout}
-                >
-                  {rawText}
-                </Text>
-              </Hyperlink>
+                <Markdown darkStyle={darkColor}>{rawText}</Markdown>
```
Comment (Ashoat) [inline native/calendar/entry.react.js:399]: Another issue I noticed in testing. If I add two line breaks after some text, the height calculation on `Markdown` gets broken. This indicates something wrong within the `Markdown` component itself

Comment (Ashoat) [reply native/calendar/entry.react.js:399]: Hmm, this issue seems to repro for messages. If I send a message `Test\n\nA` it turns into `TestA`. I think we tested this, so I suspect one of our later changes broke it. I'll try to investigate this as well...

Comment (Ashoat) [reply native/calendar/entry.react.js:399]: I finished the investigation and updated #67. Check out my comment there, and make sure to rebase this PR against the updated `feature/message_formatting` branch!

File: native/calendar/entry.react.js [hunk 5]
```diff
@@ -390,13 +390,15 @@ class InternalEntry extends React.Component<Props, State> {
...
-            <View>
-              <View style={[heightStyle, this.props.styles.textContainer]} />
+            <View style={heightStyle} />
+            <View style={this.props.styles.textContainer}>
               <Text
                 style={[this.props.styles.text, textStyle]}
                 onLayout={this.onTextLayout}
               >
-                <Markdown darkStyle={darkColor}>{rawText}</Markdown>
+                <Markdown textStyle={textStyle} useDarkStyle={darkColor}>
```
Comment (KatPo) [inline native/calendar/entry.react.js:399]: This style is not the right one to use, it's temporary just to make this commit

File: native/ios/Podfile.lock [hunk 6]
```diff
@@ -286,8 +286,6 @@ PODS:
   - react-native-ffmpeg/min-lts (0.4.4):
     - mobile-ffmpeg-min (= 4.3.1.LTS)
     - React
-  - react-native-in-app-message (1.0.2):
```
Comment (Ashoat) [inline native/ios/Podfile.lock:289]: This is strange - why is `react-native-in-app-message` removed here?

File: native/package.json [hunk 7]
```diff
@@ -61,7 +61,6 @@
     "react-native-floating-action": "^1.19.1",
     "react-native-fs": "2.15.2",
     "react-native-gesture-handler": "^1.6.1",
-    "react-native-hyperlink": "git+https://git@github.com/ashoat/react-native-hyperlink.git#both",
```
Comment (Ashoat) [inline native/package.json:64]: One more thing I realized - we should `git rm patches/react-native-hyperlink+0.0.12.patch`

File: native/ios/Podfile.lock (hunk 6 above)
Comment (KatPo) [reply native/ios/Podfile.lock:289]: I was also confused, but it happened after I did `pushd native && yarn remove react-native-hyperlink && popd && yarn cleaninstall`

Comment (KatPo) [reply native/ios/Podfile.lock]: Some weird bug I guess, I had to run `yarn clean install` couple of times and it's back now

**Comment by Ashoat** [comment]
Awesome!! Let's merge this into `feature/message_formatting`
